wgengine/router/router_linux.go

```diff
@@ -467,6 +521,16 @@ func (r *linuxRouter) setNetfilterMode(mode preftype.NetfilterMode) error {
 			if err := r.nfr.AddBase(r.tunname); err != nil {
 				return err
 			}
+			if r.magicsockPort4 != 0 {
+				if err := r.nfr.AddMagicsockPortRule(r.magicsockPort4, "udp4"); err != nil {
+					return err
```

Comment:
In a few places within the PR we are doing return err. I haven't pulled this locally yet to have a click through - but do all these errors get surfaced with an indication of the function that the error is happening in? That can really help narrow down the issue when it comes to looking through logs.